src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,18 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer {
```

Comment:
The `TruckProducer` class should be parameterized as `TruckProducer implements MachineProducer<Truck>` to ensure type safety and prevent the creation of invalid implementations.